run.go

```diff
@@ -0,0 +1,166 @@
+	"github.com/bitrise-io/go-steputils/v2/ruby"
+	"github.com/bitrise-io/go-utils/fileutil"
+	"github.com/bitrise-io/go-utils/pathutil"
+	"github.com/bitrise-io/go-utils/v2/command"
+	"github.com/bitrise-io/go-xcode/appleauth"
+)
+
+// RunOpts ...
+type RunOpts struct {
+	WorkDir         string
+	AuthCredentials appleauth.Credentials
+	LaneOptions     []string
+	UseBundler      bool
+	GemVersions     gemVersions
+	EnableCache     bool
+}
+
+// Run ...
+func (f FastlaneRunner) Run(opts RunOpts) error {
+	// Run fastlane
+	f.logger.Println()
+	f.logger.Infof("Run Fastlane")
+
+	var envs []string
+	authEnvs, err := FastlaneAuthParams(opts.AuthCredentials)
+	if err != nil {
+		return fmt.Errorf("Failed to set up Fastlane authentication parameters: %v", err)
+	}
+	var globallySetAuthEnvs []string
+	for envKey, envValue := range authEnvs {
+		if _, set := os.LookupEnv(envKey); set {
+			globallySetAuthEnvs = append(globallySetAuthEnvs, envKey)
+		}
+
+		envs = append(envs, fmt.Sprintf("%s=%s", envKey, envValue))
+	}
+	if len(globallySetAuthEnvs) != 0 {
+		f.logger.Warnf("Fastlane authentication-related environment varibale(s) (%s) are set, overriding.", globallySetAuthEnvs)
+		f.logger.Infof("To stop overriding authentication-related environment variables, please set Bitrise Apple Developer Connection input to 'off' and leave authentication-related inputs empty.")
+	}
+
+	buildlogPth := ""
+	if tempDir, err := pathutil.NormalizedOSTempDirPath("fastlane_logs"); err != nil {
+		f.logger.Warnf("Failed to create temp dir for fastlane logs, error: %s", err)
+	} else {
+		buildlogPth = tempDir
+		envs = append(envs, "FL_BUILDLOG_PATH="+buildlogPth)
+	}
+
+	name := "fastlane"
+	args := opts.LaneOptions
+	options := &command.Opts{
+		Stdout: os.Stdout,
+		Stderr: os.Stderr,
+		Dir:    opts.WorkDir,
+		Env:    append(os.Environ(), envs...),
+	}
+	var cmd command.Command
+	if opts.UseBundler {
+		cmd = f.rbyFactory.CreateBundleExec(name, args, opts.GemVersions.bundler.Version, options)
+	} else {
+		cmd = f.rbyFactory.Create(name, args, options)
+	}
+
+	f.logger.Donef("$ %s", cmd.PrintableCommandArgs())
+
+	deployDir := os.Getenv("BITRISE_DEPLOY_DIR")
+	if deployDir == "" {
+		f.logger.Warnf("No BITRISE_DEPLOY_DIR found")
+	}
+	deployPth := filepath.Join(deployDir, "fastlane_env.log")
+
+	if fastlaneErr := cmd.Run(); fastlaneErr != nil {
+		var exitErr *exec.ExitError
+		if errors.As(fastlaneErr, &exitErr) {
+			f.logger.Warnf("Fastlane command failed with exit status %d (%s): %w", exitErr.ExitCode(), cmd.PrintableCommandArgs(), errors.New("check the command's output for details"))
+		} else {
+			f.logger.Warnf("Fastlane command failed (%s)", cmd.PrintableCommandArgs())
+		}
+		f.logger.Warnf("If you want to send an issue report to fastlane (https://github.com/fastlane/fastlane/issues/new), you can find the output of fastlane env in the following log file:")
+		f.logger.Println()
+		f.logger.Infof(deployPth)
+		f.logger.Println()
```

Comment:
We print the command error later already, so instead, I suggest this:

```suggestion
		f.logger.Println()
		f.logger.Warnf(`Running Fastlane failed. If you want to send an issue report to Fastlane (https://github.com/fastlane/fastlane/issues/new),
you can find the output of fastlane env in the following log file: %s`, deployPth)
```